src/lib/components/AccessRightField.js

```diff
@@ -1,53 +1,110 @@
 // This file is part of React-Invenio-Deposit
-// Copyright (C) 2020 CERN.
-// Copyright (C) 2020 Northwestern University.
+// Copyright (C) 2020-2021 CERN.
+// Copyright (C) 2020-2021 Northwestern University.
 //
 // React-Invenio-Deposit is free software; you can redistribute it and/or modify it
 // under the terms of the MIT License; see LICENSE file for more details.
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import _get from 'lodash/get';
 
-import {
-  ErrorLabel,
-  FieldLabel,
-  RadioField,
-  GroupField,
-} from 'react-invenio-forms';
-import { Card, Form } from 'semantic-ui-react';
-import { Field } from 'formik';
+import { FieldLabel } from 'react-invenio-forms';
+import { Card, Divider, Form } from 'semantic-ui-react';
+
+import { Embargo, Embargoed, EmbargoedFiles, Public, Restricted, RestrictedFiles } from "./Access";
+
+
+class Protection {
+  static create(
+    metadataPublic,
+    filesPublic,
+    hasFiles,
+    embargo
+  ) {
+
+    if (metadataPublic) {
+      if (hasFiles) {
+        if (filesPublic) {
+          return new Public(hasFiles, embargo);
+        } else if (embargo.is(Embargo.APPLIED)) {
+          return new EmbargoedFiles(embargo);
+        } else {
+          return new RestrictedFiles(embargo);
+        }
+      }
+
+      if (!hasFiles) {
+        return new Public(hasFiles, embargo);
+      }
+    }
+
+    if (!metadataPublic) {
+      if (embargo.is(Embargo.APPLIED)) {
+        return new Embargoed(hasFiles, embargo);
+      } else {
+        return new Restricted(hasFiles, embargo);
+      }
+    }
+  }
+}
+
 
 export class AccessRightField extends Component {
   /** Top-level Access Right Component */
 
   render() {
     const { fieldPath, label, labelIcon, options } = this.props;
 
+    // External
```

Comment:
This is being changed in the subsequent draft PR https://github.com/inveniosoftware/react-invenio-deposit/pull/216 by the Formik field, so it can be left temporarily.